include/sea_dsa/support/Brunch.hh

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "llvm/ADT/Twine.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "sea_dsa/support/Debug.h"
+
+#include <sys/resource.h>
+
+#include <cassert>
+#include <memory>
+
+/// Brunch stats can be enabled by passing --log=brunch to a sea-dsa binary.
+/// Brunch progress can be enabled by passing --log=brunch to a sea-dsa binary.
+
+#define SEA_DSA_BRUNCH_STAT(...)                                               \
+  LOG("brunch", sea_dsa::PrintBrunchStat(__VA_ARGS__))
+#define SEA_DSA_BRUNCH_PROGRESS(...)                                           \
+  LOG("brunch-progress", sea_dsa::PrintBrunchProgress(__VA_ARGS__))
+
+namespace sea_dsa {
+template <typename T> void PrintBrunchStat(llvm::Twine key, const T &value) {
+  llvm::errs() << "BRUNCH_STAT " << key << " " << value << "\n";
+}
+
+template <typename T>
+void PrintBrunchProgress(llvm::Twine key, const T &value, const T &max = "") {
+  llvm::errs() << "BRUNCH_PROGRESS " << key << " " << value << " / " << max
+               << "\n";
+}
+
+long GetCurrentMemoryUsageKb();
+
+class BrunchTimer {
```

Comment:
We already have a stopwatch and scoped timers. Why is this necessary? If you don't like how we measure time, change it there.

Reply:
it's in seahorn, not in sea_dsa

Reply:
at the moment we just copy code instead of creating new and different one